### DISCUSSION TITLE
Regarding the Management scope assignment

### DISCUSSION THREAD

**Discussion by mrunmayeeindap** [gettek/terraform-azurerm-policy-as-code#78] [Q&A]
Hello,

Is it possible to assign the policy for the Management group name with the spaces or can we use the Tenant Root Management level in data block file. I tried using it but it is throwing error. Any suggestion?

Thanks!
Mona

**Comment by gettek** [comment]
Hi @mrunmayeeindap, could you share your data call here please

**Reply by mrunmayeeindap** [reply]
Hello @gettek ,

Thank you so much for the reply. Please find the below data call details:

```hcl
# Org Management Group
data "azurerm_management_group" "org" {
  name =  "My Management group - RND"
}
```
Actually I want to apply policy definition at Management group level. Because of the space it is throwing error.


Thanks!
Mona

**Reply by gettek** [reply]
Seems related to an old [API issue](https://github.com/hashicorp/terraform-provider-azurerm/issues/6757) which should now be resolved, have you tried using `display_name` instead? If this persist I would simply change the MG `name` to something without whitespaces and keep the `display_name` should you wish it to remain the same.

**Reply by mrunmayeeindap** [reply]
@gettek Thank you for the response. Also one more query is it possible to use System Managed identity for Deployifnotexist mode instead of using the user defined identity.

Thanks!
Mona

**Reply by gettek** [reply]
Hi @mrunmayeeindap, a system assigned managed identify gets automatically created for policies with either DINE or Modify effect.


Role assignments are also created based off role definition Ids stored in the Policy Definition, or by explicitly providing `role_definition_ids` to the module.

Remediation tasks are also automatically created, both can be prevented with the `skip_role_assigment` and `skip_remediation` flags set to `true`.

Please review module readme's for better clarification on acceptable inputs

**Reply by mrunmayeeindap** [reply]
@gettek Hello, Thank you so much its working now.

Also one more query i have,  while we assign any assignment through policy as a code we are not getting the name of the assigner. Like we get in portal. Any idea what can be done here can we get the Assigned by details.

<img width="890" alt="image" src="https://github.com/gettek/terraform-azurerm-policy-as-code/assets/57096562/34e5e34d-5f18-4218-9a34-3f51eca29ae6">


Thanks!
Mona

**Reply by gettek** [reply]
Hi @mrunmayeeindap, is there a property within the assignment Json named `"systemData"` if not please check the metadata is populated with `"createdBy"`/`"updatedBy"` properties
